src/shelly_temp_sensor_ow.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Shelly-HomeKit Contributors
+ * All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "shelly_temp_sensor.hpp"
+
+#include <vector>
+
+#include <mgos_onewire.h>
+
+#include "mgos_timers.hpp"
+
+namespace shelly {
+
+class OWTempSensor;
+
+class Onewire {
+ public:
+  Onewire(int pin_in, int pin_out);
+  virtual ~Onewire();
```

Comment:
doesn't need to be virtual, since the class doesn't inherit from and it not inherited by any othe rclass.